src/components/subscriptions/addons/EditAddon.js

```diff
@@ -0,0 +1,222 @@
+import React, { useState, useCallback, useEffect } from "react";
+
+import { useQuery, useMutation, useQueryClient } from "react-query";
+import { useTranslation } from "i18n";
+
+import buildID from "components/utils/DebugIDUtil";
+import { Button, MenuItem } from "@material-ui/core";
+import DEDialog from "components/utils/DEDialog";
+import { FastField, Field, Form, Formik } from "formik";
+import FormCheckbox from "components/forms/FormCheckbox";
+import FormNumberField from "components/forms/FormNumberField";
+import FormTextField from "components/forms/FormTextField";
+import { nonEmptyField, nonZeroValue } from "components/utils/validations";
+import ErrorTypographyWithDialog from "components/error/ErrorTypographyWithDialog";
+
+import ids from "../ids";
+import { formatAddonSubmission, mapPropsToValues } from "./formatters";
+
+import {
+    getResourceTypes,
+    postAddon,
+    AVAILABLE_ADDONS_QUERY_KEY,
+    RESOURCE_TYPES_QUERY_KEY,
+} from "serviceFacades/subscriptions";
+import { announce } from "components/announcer/CyVerseAnnouncer";
+
+function EditAddonDialog(props) {
+    const { open, onClose, parentId } = props;
+    const { t } = useTranslation("subscriptions");
+    const queryClient = useQueryClient();
+    const resourceTypesCache = queryClient.getQueryData(
+        RESOURCE_TYPES_QUERY_KEY
+    );
+
+    const [resourceTypes, setResourceTypes] = useState();
+    const [resourceTypesQueryEnabled, setResourceTypesQueryEnabled] =
+        useState();
+    const [postAddonError, setPostAddonError] = useState();
+
+    const preProcessResourceTypes = useCallback(
+        (data) => {
+            if (data?.result?.length > 0) {
+                let types = [];
+                data.result.forEach((type) => types.push(type));
+                setResourceTypes(types);
```

Comment:
I think this could also be accomplished with the spread syntax:
```suggestion
                let types = data.result;
                setResourceTypes([...types]);
```